docker/cli
# cli/command/container: implement `docker run --annotation`

**- What I did**
Implemented `docker run --annotation`.

For moby/moby#45025 (Docker v24, API v1.43).

`docker run --annotation foo=bar` is similar to `podman run --annotation foo=bar`, however, unlike Podman, Docker implementation also accepts an annotation with an empty value. (`docker run --annotation foo`)


**- How I did it**

Added the `--annotation` flag as an `*opts.MapOpts`.

**- How to verify it**

Run `docker run --annotation foo=bar`, and run `docker inspect` to confirm that the annotation is set.

**- Description for the changelog**

cli/command/container: implement `docker run --annotation`

**- A picture of a cute animal (not mandatory but encouraged)**
🐧
